server/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# create base image with all packages, proper entrypoint, and directories created
+FROM node:dubnium-alpine AS base
+
+# install any packages we need from apt here
+RUN apk add --update dumb-init
+
+# set entrypoint to `dumb-init` as it handles being pid 1 and forwarding signals
+# so that you dont need to bake that logic into your node app
+ENTRYPOINT ["dumb-init", "--"]
+
+# all of our code will live in `/app`
+WORKDIR /app
+
+# using the base image, create an image containing all of our files
+# and dependencies installed, devDeps and test directory included
+FROM base AS dependencies
+
+COPY package*.json ./
+RUN npm set progress=false \
+    && npm config set depth 0 \
+    && npm i
+COPY .eslintrc.json .
+COPY ./server ./server
+COPY ./test ./test
+COPY ./lib ./lib
+
+# if you have any build scripts to run, like for the `templated-site` flavor
+# uncomment and possibly modify the following RUN command:
+# RUN npm run build
+# keeping all of the bash commands you can within a single RUN is generally important,
+# but for this case it's likely that we want to use the cache from the prune which will
+# change infrequently.
+
+# test running image using all of the files and devDeps
+FROM dependencies AS test
+USER node
+ENV NODE_ENV=test
```

Comment:
I believe lab sets the `NODE_ENV` to `'test'`.  Does that affect this?

Reply:
This is more of a CYA line. Not all test runners explicitly set to `test`, and i've seen issues with some that _say_ they do, but then dont and cause issues.